### PR TITLE
Move embedding test into the test directory

### PR DESCRIPTION
**Pull request by ararslan**
In preparation for excising all of the examples. This is the only one that's useful as a test.

### CODE REVIEW

**Comment by ararslan** [comment]
AppVeyor isn't working at all, so it would be great if someone who understands Windows/Cygwin could make sure that my change to the AppVeyor configuration makes sense.

File: .travis.yml [hunk 1]
```diff
@@ -148,7 +148,7 @@ script:
     - popd
     # test that the embedding code works on our installation
     - mkdir /tmp/embedding-test &&
-        make check -C /tmp/julia/share/doc/julia/examples/embedding \
+        make check -C embedding \
```
Comment (staticfloat) [inline .travis.yml]: Why did you get rid of the path here?  I don't think this works; you still need to give it the path, otherwise it can't find it.

Comment (ararslan) [reply .travis.yml]: Yeah I don't know why I did that. :upside_down_face:

**Comment by ararslan** [comment]
The CI failures are unrelated.

**Comment by ararslan** [comment]
@staticfloat Want to take another look?

**Comment by staticfloat** [comment]
@ararslan I merged this too early apparently; [this file](https://github.com/JuliaLang/julia/blob/master/examples/Makefile#L13) needs an update for this to succeed.

**Pull request merged**